Lib/multiprocessing/pool.py

```diff
@@ -429,15 +440,28 @@ def _map_async(self, func, iterable, mapper, chunksize=None, callback=None,
         )
         return result
 
+    def _wait_for_updates(self, timeout):
+        task_queue_sentinels = [self._outqueue._reader.fileno(),
+                                self._outqueue._writer.fileno()]
+        self_notifier_sentinels = [self._change_notifier._reader.fileno()]
+        worker_sentinels = [worker.sentinel for worker in self._pool]
+        sentinels = [*task_queue_sentinels,
+                     *worker_sentinels,
+                     *self_notifier_sentinels]
+        wait(sentinels, timeout=timeout)
+
     @staticmethod
     def _handle_workers(pool):
         thread = threading.current_thread()
 
         # Keep maintaining workers until the cache gets drained, unless the pool
         # is terminated.
-        while thread._state == RUN or (pool._cache and thread._state != TERMINATE):
```

Comment:
Why did you change this?

Reply:
@pitrou Because we now have notifiers for these events and I want to uncouple the loop with the checks so we could move them in the inner functions if needed. I think the break conditions are a bit more clear also.

I am happy to change it back if you prefer, thought.

Reply:
Hummmm, Sorry, I think I am missing a commit here.